### DISCUSSION TITLE
Modify callback: think about how to deal with relation between multiple fields

### DISCUSSION THREAD

**Discussion by marwie** [needle-tools/custom-timeline-editor#9] [Ideas]
``` 
kann ich für meinen type überschreiben wann der in screenspace getroffen wird?
also Beispiel: ich hab Vector3 aber rechne in "anderer Space" also müsste ich dem Tool sagen wie es von data > world > data rechnen kann
oder anderes beispiel: ich hab Vector3 from; Vector3 to; und will dass zB beim Malen von "thickness" egal ist ob ich jetzt anfang, ende, oder mitte der Linie treffe
gerade muss ich mir merken und/oder über Gizmo malen wo "vorne" ist weil ich nur da malen kann
```

**Comment by hybridherbst** [comment]
hätt glaub eher gedacht du hast nen extra interface für "get weight for tool"
wenn ich Lines baue dann würd ich das implementieren, vll sowas wie

```
: ICustomToolWeight

float GetCustomToolWeight(ToolType tool, ToolSettings settings) {
  return DistanceToLine(myData.from, myData.to, myData.thickness, settings.position, settings.radius);
}
```

**Reply by marwie** [reply]
Wäre ne Möglichkeit, kann ich die Tage mal probieren :)

**Reply by marwie** [reply]
Probier nächstes mal das zu implementieren:

![image](https://user-images.githubusercontent.com/5083203/140832718-4f6e199e-12a4-471d-a07c-0a698c0404c5.png)

**Reply by marwie** [reply]
Dann musst du aber alles selber berechnen :) is aber kein problem wenn du in deinem type bist :)

**Reply by marwie** [reply]
Schöner wäre aber automatisch ;)

**Reply by hybridherbst** [reply]
![image](https://user-images.githubusercontent.com/2693840/140835595-31449cc9-0ca2-4835-a937-b84c6ac39e78.png)

**Reply by marwie** [reply]
both: cast caller
